### ISSUE TITLE
contrast ratios

### ISSUE THREAD

**Issue by MrLoh** [hsluv/hsluv#75]
My understanding was that the definition of the L value value of HSLuv corresponds to the [relative luminance as defined in the w3c standard](https://www.w3.org/TR/2008/REC-WCAG20-20081211/#relativeluminancedef) when it comes to calculating [w3c contrast ratios](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-contrast.html) as [proven here](https://github.com/hsluv/hsluv/blob/master/math/w3c.mac). But when I calculate the contrast ration with the javascript implementation I get quite different contrast ratios between the 2 methods. What am I missing?

HSLuv based contrast ratio
```js
const contrast = (color1, color2) => {
  const [h1, s1, l1] = hexToHsluv(color1);
  const [h1, s1, l2] = hexToHsluv(color2);
  return l1 > l2 ? (l1 / 100 + 0.05) / (l2 / 100 + 0.05) : (l2 / 100 + 0.05) / (l1 / 100 + 0.05);
};
```

sRGB based contrast ratio
```js
const contrast = (color1, color2) => {
  const f = (v) => (v <= 0.03928 ? v / 12.92 : Math.pow((v + 0.055) / 1.055, 2.4));
  const [r1, g1, b1] = hsluvToRgb(hexToHsluv(color1));
  const l1 = 0.2126 * f(r1) + 0.7152 * f(g1) + 0.0722 * f(b1);
  const [r2, g2, b2] = hsluvToRgb(hexToHsluv(color2));
  const l2 = 0.2126 * f(r2) + 0.7152 * f(g2) + 0.0722 * f(b2);
  return l1 > l2 ? (l1 + 0.05) / (l2 + 0.05) : (l2 + 0.05) / (l1 + 0.05);
};
```

**Comment by MrLoh** [comment]
After a bit of research I learned that the WCAG standard is basically inaccurate (see https://github.com/w3c/wcag/issues/695) and a Lab based color contrast definition is more accurate. I verified that the hsluv lightness channel models the SAPC contrast closely and a contrast of 50 in that channel would correspond to the SAPC 80% contrast threshold (see https://codesandbox.io/s/contrast-vs-lab-lightness-9lsdw). 

In the end I abandoned this library though and used chroma-js for generating color schemes, because the sudden chroma shifts in hsluv were becoming problematic in some cases.
